service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/EhrResendControllerTest.java

```diff
@@ -0,0 +1,269 @@
+                                                           .conversationClosed(FIVE_DAYS_AGO)
+                                                           .errors(List.of())
+                                                           .messageRef(ehrMessageRef)
+                                                           .received(FIVE_DAYS_AGO)
+                                                           .rootId(generateRandomUppercaseUUID())
+                                                           .build());
+        ehrExtractStatus.setEhrRequest(EhrExtractStatus.EhrRequest.builder().nhsNumber(NHS_NUMBER).build());
+
+        doReturn(Optional.of(ehrExtractStatus)).when(ehrExtractStatusRepository).findByConversationId(CONVERSATION_ID);
+
+        var response = ehrResendController.scheduleEhrExtractResend(CONVERSATION_ID);
+
+        JsonNode rootNode = objectMapper.readTree(response.getBody());
+
+        assertResponseHasExpectedOperationOutcome(rootNode, INTERNAL_SERVER_ERROR, DIAGNOSTICS_MSG, ISSUE_CODE_BUSINESS_RULE);
+        assertEquals(HttpStatus.FORBIDDEN, response.getStatusCode());
+    }
+
+    @Test
+    void When_AnEhrExtractDoesNotExist_Expect_RespondsWith404() throws JsonProcessingException {
+
+        var details = new CodeableConcept();
+        var codeableConceptCoding = new Coding();
+        codeableConceptCoding.setSystem(GPCONNECT_ERROR_OR_WARNING_CODE);
+        codeableConceptCoding.setCode("INVALID_IDENTIFIER_VALUE");
+        details.setCoding(List.of(codeableConceptCoding));
+        var diagnosticsMsg = "Provide a conversationId that exists and retry the operation";
+
+        doReturn(Optional.empty()).when(ehrExtractStatusRepository).findByConversationId(CONVERSATION_ID);
+
+        var response = ehrResendController.scheduleEhrExtractResend(CONVERSATION_ID);
+
+        JsonNode rootNode = objectMapper.readTree(response.getBody());
+
+        assertResponseHasExpectedOperationOutcome(rootNode, INVALID_IDENTIFIER_VALUE, diagnosticsMsg, ISSUE_CODE_VALUE);
+        assertEquals(HttpStatus.NOT_FOUND, response.getStatusCode());
+    }
+
+    private void assertResponseHasExpectedOperationOutcome(JsonNode rootNode, String serverErrMsg,
+                                                           String diagnosticsMsg, String issueCode) {
+        var coding = rootNode.path("issue").get(0).path("details").path("coding").get(0);
+        assertEquals(serverErrMsg, coding.path("code").asText());
+        assertEquals("error", rootNode.path("issue").get(0).path("severity").asText());
+        assertEquals(issueCode, rootNode.path("issue").get(0).path("code").asText());
+        assertEquals(GPCONNECT_ERROR_OR_WARNING_CODE, coding.path("system").asText());
+        assertEquals(URI_TYPE, rootNode.path("meta").path("profile").get(0).asText());
+        assertEquals(diagnosticsMsg, rootNode.path("issue").get(0).path("diagnostics").asText());
```

Comment:
again should be wrapped in assertAll, probably should be using assertJ test methods for readability